docs/vfio-user.rst

```diff
@@ -513,12 +537,44 @@ The migration capability contains the following name/value pairs:
 |        |        | between the client and the server is used.    |
 +--------+--------+-----------------------------------------------+
 
+The ``twin_socket`` capability object holds these name/value pairs:
+
++----------+---------+--------------------------------------------------------+
+| Name     | Type    | Description                                            |
++==========+=========+========================================================+
+| enable   | boolean | Indicates whether the client wants to enable           |
+|          |         | twin-socket mode. Optional, defaults to false, only    |
+|          |         | valid in the request message.                          |
++----------+---------+--------------------------------------------------------+
+| fd_index | number  | Specifies an index in the file descriptor array        |
+|          |         | included with the message. The designated file         |
+|          |         | descriptor is a socket which is to be used for the     |
+|          |         | server-to-client command channel. Optional, only valid |
+|          |         | in the reply message.                                  |
++----------+---------+--------------------------------------------------------+
+
 Reply
 ^^^^^
 
 The same message format is used in the server's reply with the semantics
 described above.
 
+If and only if the client has requested to enable twin-socket mode by setting
```

Comment:
Yes. Although as worded, the server would just omit the `fd_index` field as opposed to setting it to -1. Or it might leave out the `twin_socket` object entirely, matching what servers that don't know `twin_socket` would do.

I can add another paragraph to capture this if you like?